src/WebUI/Workloads/WorkloadsStore.ts

```diff
@@ -80,24 +84,40 @@ export class WorkloadsStore extends StoreBase {
     }
 
     private _setReplicaSetsList = (replicaSetList: V1ReplicaSetList): void => {
-        this._state.replicaSetList = replicaSetList;
-        this.emit(WorkloadsEvents.ReplicaSetsFetchedEvent, this);
+        const _stateItems = this._state.replicaSetList ? this._state.replicaSetList.items : [];
+        const _replicaItems = replicaSetList ? replicaSetList.items : [];
+        const isEquals = Utils.isDeepEquals(_stateItems, _replicaItems);
+        if (!isEquals || !this._state.replicaSetList) {
+            this._state.replicaSetList = replicaSetList;
+            this.emit(WorkloadsEvents.ReplicaSetsFetchedEvent, this);
+        }
         if (this._state.replicaSetList && this._state.replicaSetList.items && this._state.replicaSetList.items.length > 0) {
             this.emit(WorkloadsEvents.WorkloadsFoundEvent, this);
```

Comment:
we have WorkloadsFoundEvent used in too many places, can we do the following, sorry if i am pushing for too many changes.

WorkloadsFoundEvent --> will be raised only when we know that deploymentlist and replicaset.
we should use some other name something like 'OtherWorkloadsFoundEvent' for stateful/daemon and orphanpods.